utils/generateAllMetricsDocumentation.py

```diff
@@ -123,40 +123,41 @@ def print_usage():
             print("Could not open file '" + filename + "'")
             bad = True
             continue
-        try:
-            y = load(s, Loader=Loader)
-        except YAMLError as err:
-            print("Could not parse YAML file '" + filename + "', error:\n" + str(err))
-            bad = True
-            continue
-
-        YAMLS.append(y)   # for later dump
-
-        # Check a few things in the yaml:
-        for attr in ["name", "help", "exposedBy", "description", "unit",\
-                     "type", "category", "complexity", "introducedIn"]:
-            if not attr in y:
-                print("YAML file '" + filename + \
-                      "' does not have required attribute '" + attr + "'")
+        if not bad:
```

Comment:
I think this is not meant. The `continue` above sorts out this file already, and we certainly do not want to skip validation of the current file if `bad` was set in a previous loop iteration...

Reply:
Forgot to delete. That was there initially.